### PR TITLE
[WIP] Bring back autosave feature

### PR DESCRIPTION
**Pull request by meganyin13**
### Summary 
Implementing cache for auto-save. Currently still has the first character lag. How to proceed?

This pull request is the first step towards implementing feature Autosave

- [ ] implemented X
- [x] fixing autosave

### Test Plan 
Focus a one-time task and try adding a new subtask. Ensure the changes automatically propagate to future view. 

### Breaking Changes  

- Database schema change (anything that changes Firestore collection structure)
- Other change that could cause problems (Detailed in notes)

### CODE REVIEW

File: frontend/src/components/Util/TaskEditors/TaskEditor/index.tsx [hunk 1]
```diff
@@ -92,16 +99,36 @@ function TaskEditor(
   const { name, tag, date, complete, inFocus } = mainTask;
   const { removeTask, onSave } = actions;
 
+  const [tempSubTask, setTempSubTask] = useState<SubTask | null>(null);
   const [subTaskToFocus, setSubTaskToFocus] = useState<TaskToFocus>(null);
 
+<<<<<<< Updated upstream
```
Comment (SamChou19815) [inline frontend/src/components/Util/TaskEditors/TaskEditor/index.tsx]: 🤔

File: frontend/src/components/Util/TaskEditors/TaskEditor/index.tsx [hunk 2]
```diff
@@ -248,15 +270,6 @@ function TaskEditor(
...
       </div>
-      <div
-        className={styles.SaveButtonRow}
-        style={diffIsEmpty(diff) ? { maxHeight: 0, padding: 0 } : undefined}
-      >
-        <span className={styles.TaskEditorFlexiblePadding} />
-        <div role="presentation" className={styles.SaveButton} onClick={onSaveClicked}>
-          <span className={styles.SaveButtonText}>Save</span>
-        </div>
-      </div>
```
Comment (SamChou19815) [inline frontend/src/components/Util/TaskEditors/TaskEditor/index.tsx]: It's unsafe to remove it completely now. Repeating task still needs it.

**Comment by meganyin13** [comment]
I have added in the cache functionality present before #245. It still does not work however I think I have narrowed down to problem to one of deciding when to call
`setTempSubTask(null)` essentially when to reset the subtask. 
Also somehow now deleting a task is broken/corrupted 🤔

**Comment by meganyin13** [comment]
From my latest commit, if you press enter/tab after typing in some text for a new subtask, the subtask will be added. Also fixed is the corrupted subtask error you get when you try to delete a task.

File: frontend/src/components/Util/TaskEditors/TaskEditor/index.tsx [hunk 3]
```diff
@@ -92,17 +96,25 @@ function TaskEditor(
...
-    }
-  };
   const onSaveClicked = (): void => {
     if (type === 'ONE_TIME') {
+      if (tempSubTask != null) {
+        addSubTask(tempSubTask);
+      }
       editTaskWithDiff(id, 'EDITING_ONE_TIME_TASK', diff);
       onSave();
+      // setPrevTempSubTask(null);
```
Comment (SamChou19815) [inline frontend/src/components/Util/TaskEditors/TaskEditor/index.tsx]: 🤔

**Comment by SamChou19815** [comment]
@meganyin13 partially fixed the losing focus issue in TaskEditor. We still need to cleanup some of the code during work session.

**Comment by meganyin13** [comment]
All changes are in #359

**Pull request closed without merging**